janaganana/settings.py.txt
from wazimap.settings import *
import os

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

#GEOS_LIBRARY_PATH = '/usr/local/lib/libgeos_c.so'
GDAL_LIBRARY_PATH = '/usr/local/lib/libgdal.so'

# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'wt08^t@ugzs4sw*qn=c*=$d+jgkqkkp4$0z98j-k5s!o2um$(n'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = ['192.168.2.115']

# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'census',
    'wazimap',
    'django.contrib.humanize',
    'django.contrib.sitemaps',
    'sass_processor',
    'leaflet',
    'jsonify',
]

MIDDLEWARE_CLASSES = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'django.middleware.cache.UpdateCacheMiddleware',
    'django.middleware.cache.FetchFromCacheMiddleware',
    'corsheaders.middleware.CorsMiddleware',
]

ROOT_URLCONF = 'janaganana.urls'
# database settings
DATABASES = {
    'default': {
        'ENGINE':   'django.db.backends.postgresql',
        'NAME':     'wazimap',
        'USER':     'wazimap',
        'PASSWORD': 'wazimap',
        'HOST':     'localhost',
        'PORT': '',
        'ATOMIC_REQUESTS': True,
    }
}

INSTALLED_APPS = ['janaganana'] + INSTALLED_APPS
#DATABASE_URL = config(
#    'DATABASE_URL', default='postgresql://factlyin:factlyin@ci-db/factlyin')
#DATABASES['default'] = dj_database_url.parse(DATABASE_URL)
#DATABASES['default']['ATOMIC_REQUESTS'] = True

# Localise this instance of Wazimap
WAZIMAP['name'] = 'Counting India'
# NB: this must be https if your site supports HTTPS.
WAZIMAP['url'] = 'http://127.0.0.1:8000'
WAZIMAP['country_code'] = 'IN'
#WAZIMAP['profile_builder'] = 'janaganana.profiles.get_census_profile'

WAZIMAP['levels'] = {
    'country': {
        'name': 'country',
        'plural': 'countries',
        'children': ['state']
    },

    'state': {
        'name':'state',
        'plural': 'states',
        'children': ['district']
    },

    'district': {
        'name':'district',
        'plural': 'districts',
        'children': []
    }
}


WAZIMAP['default_geo_version'] = '2011'
WAZIMAP['comparative_levels'] = ['country', 'state', 'district']

WAZIMAP['geometry_data'] = {'2011': {
  'country':  'geo/country.topojson',
  'state':    'geo/state.topojson',
  'district': 'geo/district.topojson',
}}

wazi_profile = os.environ.get('WAZI_PROFILE', 'census')
WAZIMAP['default_profile']    = wazi_profile
WAZIMAP['profile_builder']    = 'janaganana.profiles.{}.get_census_profile'.format(wazi_profile)

# Custom timeseries profile
timeseries_profile = os.environ.get('WAZI_PROFILE', 'timeseries')
WAZIMAP['timeseries_profile'] = timeseries_profile
WAZIMAP['timeseries_profile_builder'] = 'janaganana.profiles.{}.get_timeseries_profile'.format(timeseries_profile)

# testing GDAL
try:
    import osgeo.gdal  # noqa
    HAS_GDAL = True
except ImportError:
    HAS_GDAL = False

#WAZIMAP['ga_tracking_id'] = 'UA-91398887-1'
WAZIMAP['twitter'] = '@factlydotin'
WAZIMAP['cache_secs'] = 0
WAZIMAP['embed_cache_secs'] = 0
WAZIMAP['map_centre'] = [20.5937, 78.9629]
WAZIMAP['map_zoom'] = 4


WSGI_APPLICATION = 'janaganana.wsgi.application'



# Password validation
# https://docs.djangoproject.com/en/1.9/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]

STATIC_URL = '/static/'
STATIC_ROOT = os.path.join(BASE_DIR, 'static-root')
STATICFILES_DIRS = [
    os.path.join(BASE_DIR, 'static')
]

STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
)

if DEBUG:
    CACHES = {
        'default': {
            'BACKEND': 'django.core.cache.backends.dummy.DummyCache',
        }
    }
else:
    CACHES = {
        'default': {
            'BACKEND': 'django.core.cache.backends.memcached.MemcachedCache',
            'LOCATION': '127.0.0.1:11211',
        }
    }

LOGGING = {
    'version': 1,
    'disable_existing_loggers': True,
    'formatters': {
        'verbose': {
            'format': '%(asctime)s %(levelname)s %(module)s %(process)d %(thread)d %(message)s'
        },
    },
    'handlers': {
        'console': {
            'level': 'DEBUG',
            'class': 'logging.StreamHandler',
            'formatter': 'verbose',
        },
        'file': {
            'level': 'INFO',
            'class': 'logging.FileHandler',
            'formatter': 'verbose',
            'filename': '/tmp/janaganana_app_debug.log',
        },
    },
    'loggers': {
        '': {
            'handlers': ['console'],
            'level': 'ERROR',
        },
        'census': {
            'level': 'DEBUG' if DEBUG else 'INFO',
        },
        'django': {
            'level': 'DEBUG' if DEBUG else 'INFO',
        },
        'django.template': {
            'level': 'ERROR',
        },
        'janaganana': {
            'handlers': ['console', 'file'],
            'level': 'DEBUG' if DEBUG else 'INFO',
        },
    }
}
SITE_ID = 1

janaganana/settings.py
from wazimap.settings import * 
import os

# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))


# SECURITY WARNING: keep the secret key used in production secret!
SECRET_KEY = 'wt08^t@ugzs4sw*qn=c*=$d+jgkqkkp4$0z98j-k5s!o2um$(n'

# SECURITY WARNING: don't run with debug turned on in production!
DEBUG = True

ALLOWED_HOSTS = ['*']

# Application definition

INSTALLED_APPS = [
    'django.contrib.admin',
    'django.contrib.auth',
    'django.contrib.contenttypes',
    'django.contrib.sessions',
    'django.contrib.messages',
    'django.contrib.staticfiles',
    'census',
    'wazimap',
    'django.contrib.humanize',
    'django.contrib.sitemaps',
    'sass_processor',
    'leaflet',
    'jsonify',
]

MIDDLEWARE_CLASSES = [
    'django.middleware.security.SecurityMiddleware',
    'django.contrib.sessions.middleware.SessionMiddleware',
    'django.middleware.common.CommonMiddleware',
    'django.middleware.csrf.CsrfViewMiddleware',
    'django.contrib.auth.middleware.AuthenticationMiddleware',
    'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
    'django.contrib.messages.middleware.MessageMiddleware',
    'django.middleware.clickjacking.XFrameOptionsMiddleware',
    'django.middleware.cache.UpdateCacheMiddleware',
    'django.middleware.cache.FetchFromCacheMiddleware',
    'corsheaders.middleware.CorsMiddleware',
]

ROOT_URLCONF = 'janaganana.urls'
# database settings
DATABASES = {
    'default': {
        'ENGINE':   'django.db.backends.postgresql',
        'NAME':     'wazimap',
        'USER':     'wazimap',
        'PASSWORD': 'wazimap',
        'HOST':     'localhost',
        'PORT': '',
        'ATOMIC_REQUESTS': True,
    }
}

INSTALLED_APPS = ['janaganana'] + INSTALLED_APPS

# Localise this instance of Wazimap
WAZIMAP['name'] = 'Counting India'
# NB: this must be https if your site supports HTTPS.
WAZIMAP['url'] = 'http://127.0.0.1:8000'
WAZIMAP['country_code'] = 'IN'
WAZIMAP['profile_builder'] = 'janaganana.profiles.get_census_profile'

WAZIMAP['levels'] = {
    'country': {
        'name': 'country',
        'plural': 'countries',
        'children': ['state']
    },

    'state': {
        'name':'state',
        'plural': 'states',
        'children': ['district']
    },

    'district': {
        'name':'district',
        'plural': 'districts',
        'children': []
    }
}


WAZIMAP['default_geo_version'] = ''
WAZIMAP['comparative_levels'] = ['country', 'state', 'district']

WAZIMAP['geometry_data'] = {'': {
  'country':  'geo/country.topojson',
  'state':    'geo/state.topojson',
  'district': 'geo/district.topojson',
}}
# testing GDAL
try:
    import osgeo.gdal  # noqa
    HAS_GDAL = True
except ImportError:
    HAS_GDAL = False

#WAZIMAP['ga_tracking_id'] = 'UA-88773672-1'
WAZIMAP['cache_secs'] = 0
WAZIMAP['embed_cache_secs'] = 0
WAZIMAP['map_centre'] = [20.5937, 78.9629]
WAZIMAP['map_zoom'] = 4


WSGI_APPLICATION = 'janaganana.wsgi.application'



# Password validation
# https://docs.djangoproject.com/en/1.9/ref/settings/#auth-password-validators

AUTH_PASSWORD_VALIDATORS = [
    {
        'NAME': 'django.contrib.auth.password_validation.UserAttributeSimilarityValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.MinimumLengthValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.CommonPasswordValidator',
    },
    {
        'NAME': 'django.contrib.auth.password_validation.NumericPasswordValidator',
    },
]

STATIC_URL = '/static/'
STATIC_ROOT = os.path.join(BASE_DIR, 'static-root')
STATICFILES_DIRS = [
    os.path.join(BASE_DIR, 'static')
]

STATICFILES_FINDERS = (
    'django.contrib.staticfiles.finders.FileSystemFinder',
    'django.contrib.staticfiles.finders.AppDirectoriesFinder',
)

if DEBUG:
    CACHES = {
        'default': {
            'BACKEND': 'django.core.cache.backends.dummy.DummyCache',
        }
    }
else:
    CACHES = {
        'default': {
            'BACKEND': 'django.core.cache.backends.memcached.MemcachedCache',
            'LOCATION': '127.0.0.1:11211',
        }
    }

LOGGING = {
    'version': 1,
    'disable_existing_loggers': True,
    'formatters': {
        'verbose': {
            'format': '%(asctime)s %(levelname)s %(module)s %(process)d %(thread)d %(message)s'
        },
    },
    'handlers': {
        'console': {
            'level': 'DEBUG',
            'class': 'logging.StreamHandler',
            'formatter': 'verbose',
        },
        'file': {
            'level': 'INFO',
            'class': 'logging.FileHandler',
            'formatter': 'verbose',
            'filename': '/tmp/janaganana_app_debug.log',
        },
    },
    'loggers': {
        '': {
            'handlers': ['console'],
            'level': 'ERROR',
        },
        'census': {
            'level': 'DEBUG' if DEBUG else 'INFO',
        },
        'django': {
            'level': 'DEBUG' if DEBUG else 'INFO',
        },
        'django.template': {
            'level': 'ERROR',
        },
        'janaganana': {
            'handlers': ['console', 'file'],
            'level': 'DEBUG' if DEBUG else 'INFO',
        },
    }
}
SITE_ID = 1
